From c16e4d5dcb543a4eee12f791da0d286aeb8bb32b Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Wed, 29 Oct 2014 16:34:28 -0400
Subject: [fs] jbd2: free bh when descriptor block checksum fails

Message-id: <1414600485-1929-39-git-send-email-lczerner@redhat.com>
Patchwork-id: 98553
O-Subject: [RHEL 7.1 PATCH v2 38/55] jbd2: free bh when descriptor block checksum fails
Bugzilla: 1156096
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: "Darrick J. Wong" <darrick.wong@oracle.com>

BZ 1156096
https://bugzilla.redhat.com/show_bug.cgi?id=1156096

BREW 8169722
https://brewweb.devel.redhat.com/taskinfo?taskID=8169722

Upstream commit 064d83892e9ba547f7d4eae22cbca066d95210ce

 Free the buffer head if the journal descriptor block fails checksum
 verification.

 This is the jbd2 port of the e2fsprogs patch "e2fsck: free bh on csum
 verify error in do_one_pass".

 Signed-off-by: Darrick J. Wong <darrick.wong@oracle.com>
 Signed-off-by: Theodore Ts'o <tytso@mit.edu>
 Reviewed-by: Eric Sandeen <sandeen@redhat.com>
 Cc: stable@vger.kernel.org
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/jbd2/recovery.c b/fs/jbd2/recovery.c
index 9b329b5..bcbef08 100644
--- a/fs/jbd2/recovery.c
+++ b/fs/jbd2/recovery.c
@@ -525,6 +525,7 @@ static int do_one_pass(journal_t *journal,
        !jbd2_descr_block_csum_verify(journal,
          bh->b_data)) {
     err = -EIO;
+    brelse(bh);
     goto failed;
    }
 
-- 
1.7.1